mcs/class/corlib/System.IO/FileStream.cs

```diff
@@ -188,6 +188,12 @@ internal FileStream (string path, FileMode mode, FileAccess access, FileShare sh
 				throw new ArgumentException ("Name has invalid chars");
 			}
 
+			// IsolatedStorage needs to keep the Name property to the default "[Unknown]"
+			if (!anonymous) {
+				path = Path.InsecureGetFullPath (path);
```

Comment:
Because it's not needed for anonymous. Path is never displayed see e.g. https://github.com/marek-safar/mono/blob/18658e8b9473ffeee53c07f956a7bc498875e0c2/mcs/class/corlib/System.IO/FileStream.cs#L1130